src/openms/source/ML/SVM/SimpleSVM.cpp

```
@@ -380,111 +161,94 @@
~    }
~    if (std::get<2>(indexes) + 1 < log2_p_.size())
~    {
~      neighbor_value += performance_[std::get<0>(indexes)][std::get<1>(indexes)][std::get<2>(indexes) + 1];
~      ++n_neighbors;
~    }
     neighbor_value /= n_neighbors; // avg. performance of neighbors
     tiebreaker.insert(make_pair(make_pair(neighbor_value, n_neighbors), i));
   }
-  const auto& indexes = best_indexes[tiebreaker.rbegin()->second]; // TODO: use begin if higher_better == false
+  // Use rbegin() for higher values (when higher_better is true) and begin() for lower values
+  const auto& indexes = best_indexes[higher_better ? tiebreaker.rbegin()->second : tiebreaker.begin()->second];
```

Cppcheck (2.10-2)

[error] Out of bounds access in expression 'best_indexes[higher_better?tiebreaker.rbegin()->second

(containerOutOfBounds)